mcc/odin/odin/snapshot_list_test.go

```diff
@@ -161,14 +161,29 @@ var listSnapshotsCases = []listSnapshotsCase{
 		},
 		instanceID: "",
 	},
+	// Instance selection
+	{
+		testCase: testCase{
+			expected: []*rds.DBSnapshot{
+				exampleSnapshot2,
+			},
+			expectedError: "",
+		},
+		name: "Two instances two snapshots",
+		snapshots: []*rds.DBSnapshot{
+			exampleSnapshot1,
+			exampleSnapshot2,
+		},
+		instanceID: "develop-rds",
+	},
 }
 
 func TestListSnapshots(t *testing.T) {
-	svc := newMockRDSClient()
 	for _, test := range listSnapshotsCases {
 		t.Run(
 			test.name,
 			func(t *testing.T) {
+				svc := newMockRDSClient()
```

Comment:
This line is moved in order to prevent the data in the mock not to be shared between different tests being run by `t.Run()`.
As a side info, `t.Run()` is running the lambda function as a test function concurrently to any other test function it is invoking. So `TestListSnapshots` will run all its test cases concurrently, reducing the test run time quite a little.